sktime/base/tests/test_base.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for BaseObject universal base class.
+
+tests in this module:
+
+    test_get_class_tags - tests get_class_tags inheritance logic
+    test_get_class_tag  - tests get_class_tag logic, incl default value
+    test_get_tags       - tests get_tags inheritance logic
+    test_get_tag        - tests get_tag logic, incl default value
+    test_set_tags       - tests set_tags logic and related get_tags inheritance
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+__author__ = ["fkiraly"]
+
+from copy import deepcopy
+
+from sktime.base import BaseObject
+from sktime.utils._testing.deep_equals import deep_equals
+
+
+# Fixture class for testing tag system
+class FixtureClassParent(BaseObject):
+
+    _tags = {"A": "1", "B": 2, "C": [], 3: "D"}
+
+
+# Fixture class for testing tag system, child overrides tags
+class FixtureClassChild(FixtureClassParent):
+
+    _tags = {"A": 42, 3: "E"}
+
+
+FIXTURE_CLASSCHILD = FixtureClassChild
+
+FIXTURE_CLASSCHILD_TAGS = {"A": 42, "B": 2, "C": [], 3: "E"}
+
+# Fixture class for testing tag system, object overrides class tags
+FIXTURE_OBJECT = FixtureClassChild()
+FIXTURE_OBJECT._tags_dynamic = {"A": 42424241, "B": 3}
+
+FIXTURE_OBJECT_TAGS = {"A": 42424241, "B": 3, "C": [], 3: "E"}
+
+
+def test_get_class_tags():
+    """Tests get_class_tags class method of BaseObject for correctness
+
+    Raises
+    ------
+    AssertError if inheritance logic in get_class_tags is incorrect
+    """
+    child_tags = FIXTURE_CLASSCHILD.get_class_tags()
+
+    msg = "Inheritance logic in BaseObject.get_class_tags is incorrect"
+
+    assert deep_equals(child_tags, FIXTURE_CLASSCHILD_TAGS), msg
+
+
+def test_get_class_tag():
```

Comment:
could we make test names more descriptive, what exactly does this test?

Reply:
`test_get_class_tag_core_logic_for_correctness` is not very descriptive, is it?

Not sure what would be a better alternative. `get_tag` has one purpose and one purpose only, similar to the others - so isn't it pretty clear what is being tested?